src/provision-cloud-resources/content.md

```diff
@@ -0,0 +1,729 @@
+
+_If Amazon RDS, Cloud SQL and Azure Database are so convenient, why not use those instead of rolling out and maintaining your database in Kubernetes?_
+
+The good news is that you can use managed services in Kubernetes.
+
+However, there is little consensus on how you should provision those resources and make them available to your apps in the cluster.
+
+There are three options you can choose from:
+
+1. **Externally provisioned** — you can create the managed resource on your cloud provider and store the details such as connection detail, username and password in Kubernetes.
+1. **Internally provisioned using Custom Resource Definitions (CRDs)** — You can map the managed resources to Kubernetes objects and have a component that creates the resources on the cloud provider.
+1. **Internally provisioned using infrastructure as code (IaC)** — You can script creating the resources. The script is stored and executed in the cluster.
+
+Let's explore the three options in more detail.
+
+## Creating Secrets manually
+
+You can provision your managed databases or message brokers manually or using a tool such as Terraform or Cloudformation.
+
+And you can pass the details such as username or password to your apps using a [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+Let's have a look at an example.
+
+You can provision an Amazon RDS database with:
+
+```terminal|command=1,2,3-9,10-11,12-13|title=bash
+export DB_USERNAME=master
+export DB_PASSWORD=secret99
+aws rds create-db-instance \
+  --allocated-storage 20 \
+  --db-instance-class db.m1.small \
+  --db-instance-identifier test-instance \
+  --engine mysql \
+  --master-username $DB_USERNAME \
+  --master-user-password $DB_PASSWORD
+aws rds wait db-instance-available \
+  --db-instance-identifier test-instance
+aws rds describe-db-instances \
+  --db-instance-identifier test-instance
+```
+
+The output of the last command is a JSON payload with the endpoint of the database.
+
+```json|highlight=10|title=aws-cli-output.json
+{
+  "DBInstances":[
+    {
+      "DBInstanceIdentifier":"test-instance",
+      "DBInstanceClass":"db.m5.large",
+      "Engine":"postgres",
+      "DBInstanceStatus":"available",
+      "MasterUsername":"master",
+      "Endpoint":{
+        "Address":"test-instance.cddqgyucel9j.eu-west-1.rds.amazonaws.com",
+        "Port":5432,
+        "HostedZoneId":"Z29XKXDKYMONMX"
+      }
+      /* ... more details */
+    }
+  ]
+}
+```
+
+You can combine username, password, connection string and the rest of the database details in a Kubernetes secret with:
+
+```terminal|command=1|title=bash
+☄apiVersion: v1
+kind: Secret
+metadata:
+  name: my-secret
+type: Opaque
+data:
+  username: bWFzdGVy # master
+  password: c2VjcmV0OTk= # secret99
+  connection_url: |
+    dGVzdC1pbnN0
+    YW5jZS5jZGRx
+    Z3l1Y2VsOWou
+    ZXUtd2VzdC0x
+    LnJkcy5hbWF6
+    b25hd3MuY29t
+    # test-instance.cddqgyucel9j.eu-west-1.rds.amazonaws.com
+```
```

Comment:
I would probably show this in plain text, the way you would define the Secret, before Kubernetes encodes the data with Base64.